src/runtime-rs/crates/resource/src/rootfs/nydus_rootfs.rs

```diff
@@ -0,0 +1,149 @@
+        let extra_options =
+            NydusExtraOptions::new(rootfs).context("failed to parse nydus extra options")?;
+        info!(sl!(), "extra_option {:?}", &extra_options);
+        let rafs_meta = &extra_options.source;
+        let (rootfs_storage, rootfs_guest_path) = match extra_options.fs_version.as_str() {
+            NYDUS_ROOTFS_V5 => {
+                let rafs_mnt =
+                    do_get_guest_share_path(HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR, cid, true);
+                let rootfs_guest_path = do_get_guest_path(ROOTFS, cid, false, false);
+                rafs_mount(
+                    h,
+                    rafs_meta.to_string(),
+                    rafs_mnt,
+                    extra_options.config.clone(),
+                    None,
+                )
+                .await
+                .context("failed to do rafs mount")?;
+                let container_share_dir = get_host_rw_shared_path(sid)
+                    .join(PASSTHROUGH_FS_DIR)
+                    .join(cid);
+                let rootfs_dir = container_share_dir.join("rootfs");
+                fs::create_dir_all(rootfs_dir).context("failed to create directory")?;
+                let snapshotdir = container_share_dir.join(SNAPSHOTDIR);
+                mount::bind_mount_unchecked(
+                    extra_options.snapshotdir.clone(),
+                    snapshotdir.clone(),
+                    true,
+                )
+                .with_context(|| {
+                    format!(
+                        "failed to bind mount {} to {:?}",
+                        extra_options.snapshotdir, snapshotdir
+                    )
+                })?;
+                let mut options: Vec<String> = Vec::new();
+                options.push(
+                    "lowerdir=".to_string()
+                        + &do_get_guest_path(
+                            HYBRID_ROOTFS_WRITABLE_LAYER_LOWER_DIR,
+                            cid,
+                            false,
+                            true,
+                        ),
+                );
+                options.push(
+                    "workdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOTDIR, "work").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push(
+                    "upperdir=".to_string()
+                        + &do_get_guest_path(
+                            format!("{}/{}", SNAPSHOTDIR, "fs").as_str(),
+                            cid,
+                            false,
+                            false,
+                        ),
+                );
+                options.push("index=off".to_string());
+                Ok((
+                    Storage {
+                        driver: KATAOVERLAYDEVTYPE.to_string(),
```

Comment:
`driver` and `fs_type` are som bit different from the golang version:

https://github.com/kata-containers/kata-containers/blob/9516286f6dd5cfd6b138810e5d7c9e01cf6fc043/src/runtime/virtcontainers/kata_agent.go#L814-L820

Can you confirm this?